backend/src/main/java/asegroup1/api/services/landregistry/LandRegistryServiceImpl.java

```diff
@@ -85,11 +90,44 @@ public LandRegistryServiceImpl() throws IOException {
             );
         }
 
-        return transactionsList;
+        return getPositionForAddresses(transactionsList);
     }
 
+    public List<Address> getPositionForAddresses(List<Address> addresses) {
```

Comment:
If the header was changed to "public <E extends Address> List<E> getPositionForAddresses(List<E> addresses) " then getTransactionsForPostCode could be changed to return List<AddressWithTransaction> by specifiing the type in transactionsList.